gtsam/hybrid/tests/testGaussianMixtureFactor.cpp

```diff
@@ -228,12 +228,17 @@ static HybridBayesNet GetGaussianMixtureModel(double mu0, double mu1,
   auto c0 = make_shared<GaussianConditional>(z, Vector1(mu0), I_1x1, model0),
        c1 = make_shared<GaussianConditional>(z, Vector1(mu1), I_1x1, model1);
 
-  auto gm = new GaussianMixture({z}, {}, {m}, {c0, c1});
-  auto mixing = new DiscreteConditional(m, "0.5/0.5");
+  KeyVector frontalKeys{z}, continuousParents;
+  DiscreteKeys discreteParents{m};
+  std::vector<GaussianConditional::shared_ptr> conditionals = {c0, c1};
+  auto gm = make_shared<GaussianMixture>(frontalKeys, continuousParents,
```

Comment:
@dellaert clang is having a hard time figuring out the constructor for the shared pointer if I use `({z}, {}, {m}, {c0, c1})`. It's so weird.

Reply:
This seems like it is an issue regarding lvalues.